devdoc/constbuffer_requirements.md

```diff
@@ -354,4 +388,62 @@ CONSTBUFFER_FROM_BUFFER_RESULT CONSTBUFFER_from_buffer(const unsigned char* sour
 
 **SRS_CONSTBUFFER_02_072: [** `CONSTBUFFER_from_buffer` shall succeed, write in `consumed` the total number of consumed bytes from `source`, write in `destination` the constructed `CONSTBUFFER_HANDLE` and return `CONSTBUFFER_FROM_BUFFER_RESULT_OK`. **]**
 
+**SRS_CONSTBUFFER_51_028: [** Shall set the state of `destination` to `CONSTBUFFER_WRITIABLE_STATE_SEAL`. **]**
+
 **SRS_CONSTBUFFER_02_073: [** If there are any failures then shall fail and return `CONSTBUFFER_FROM_BUFFER_RESULT_ERROR`. **]**
+
+
+### CONSTBUFFER_create_writiable_handle
+
+```c
+ CONSTBUFFER_HANDLE CONSTBUFFER_create_writable_handle (uint32_t size)
+```
+
+`CONSTBUFFER_create_writiable_handle` construct a new `CONSTBUFFER_HANDLE` of size `size` and state `CONSTBUFFER_WRITABLE_STATE_OPEN`.`CONSTBUFFER_HANDLE`supports two state `CONSTBUFFER_WRITABLE_STATE_SEAL` and `CONSTBUFFER_WRITABLE_STATE_OPEN`.If the state is `CONSTBUFFER_WRITABLE_STATE_OPEN` then buffers of `CONSTBUFFER_HANDLE` can be overridden (the same buffer can be reused to fill).
+
+**SRS_CONSTBUFFER_51_001: [** If `size` is 0, then CONSTBUFFER_create_writiable_handle shall fail and return NULL. **]**
+
+**SRS_CONSTBUFFER_51_002: [** `CONSTBUFFER_create_writiable_handle` shall allocate memory for the `CONSTBUFFER_HANDLE` **]**
+
+**SRS_CONSTBUFFER_51_003: [** If any error occurs, `CONSTBUFFER_create_writiable_handle` shall fail and return NULL. **]**
+
+**SRS_CONSTBUFFER_51_004: [** `CONSTBUFFER_create_writiable_handle` shall set the ref count of the newly created `CONSTBUFFER_HANDLE` to 1. **]**
+
+**SRS_CONSTBUFFER_51_005: [** `CONSTBUFFER_create_writiable_handle` shall set the state of the newly created `CONSTBUFFER_HANDLE` to `CONSTBUFFER_WRITABLE_STATE_OPEN`. **]**
+
+**SRS_CONSTBUFFER_51_006: [** `CONSTBUFFER_create_writiable_handle` shall store the `storage` and `size` to the newly created const buffer. **]**
```

Comment:
>`storage` [](http://example.com/codeflow?start=84&length=9)

storage doe snot exist here in the specs. I suspect you are referring that internally the const buffer should set its alias to wherever the bytes are allocated. But I think you don't actually have to spec that at all, it is an implementation detail. Tests will fail if we don't do that :-).

Reply:
Removed .